commands.c

```diff
@@ -942,19 +942,20 @@ void mutt_display_address(struct Envelope *env)
 
 /**
  * set_copy_flags - Set the flags for a message copy
- * @param[in]  e       Email
- * @param[in]  decode  If true, decode the message
- * @param[in]  decrypt If true, decrypt the message
- * @param[out] cmflags Flags, see #CopyMessageFlags
- * @param[out] chflags Flags, see #CopyHeaderFlags
+ * @param[in]  e               Email
+ * @param[in]  transform_opt   decode/decrypt/none
+ * @param[out] cmflags         Flags, see #CopyMessageFlags
+ * @param[out] chflags         Flags, see #CopyHeaderFlags
  */
-static void set_copy_flags(struct Email *e, bool decode, bool decrypt,
+static void set_copy_flags(struct Email *e, enum MessageTransformOpt transform_opt,
                            CopyMessageFlags *cmflags, CopyHeaderFlags *chflags)
 {
   *cmflags = MUTT_CM_NO_FLAGS;
   *chflags = CH_UPDATE_LEN;
 
-  if ((WithCrypto != 0) && !decode && decrypt && (e->security & SEC_ENCRYPT))
+  enum MessageTransformOpt transform_opt_local = transform_opt;
```

Comment:
I think the logic is broken here: the old code was overwriting the `decode` variable where you are now writing into the `_local` version. Did you perhaps intend to `transform_opt |= TRANSFORM_DECODE` ?

Reply:
I saw that. Rewriting function arguments is a questionable thing to do IMHO. Also it created a situation where `decrypt` and `decode` could be true at the same time...